ibis/expr/api.py

```diff
@@ -2308,6 +2314,9 @@ def _interval_property(target_unit):
     __add__=_interval_add,
     add=_interval_add,
 
+    __sub__=_interval_sub,
```

Comment:
I was getting fancy combining interval arguments in my `trailing_time_window` and saw that I could not subtract intervals. Implemented as a bonus!

Reply:
This is great! Did you write a test for this on both the expression construction/validation side and on the compilation side? If it isn't strictly necessary for this PR, then let's do it in a follow up so there aren't orthogonal changes in this PR.

Reply:
I've opened a new PR for this: #1489

Reply:
Cool, thanks for doing that. Just merged it. If you rebase on top of master (or merge it in) you'll pick it up and then I'll give this PR another pass. Pretty close to merge!